ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -613,13 +614,22 @@ private boolean isEligibleForCompaction(CompactionInfo ci,
         return false;
       }
 
-      if (isNoAutoCompactSet(t.getParameters())) {
-        LOG.info("Table " + tableName(t) + " marked " + hive_metastoreConstants.TABLE_NO_AUTO_COMPACT +
-            "=true so we will not compact it.");
+      Map<String, String> dbParams = computeIfAbsent(ci.dbname, () -> resolveDatabase(ci)).getParameters();
+      if (isNoAutoCompactSet(dbParams, t.getParameters())) {
+        if (MetaStoreUtils.IS_DB_LEVEL_CONFIG) {
```

Comment:
idea is good , but not the impl. There could be multiple dbs, sharing through static filed in multiple threads is not a good idea. What if you simple use
````
if (getNoAutoCompact(dbParameters)){
skipDBs.add(ci.dbname);
} else {
skipTables.add(ci.getFullTableName());
}
````